pkg/helm/actions/install_chart.go

```diff
@@ -109,3 +109,88 @@ func InstallChart(ns, name, url string, vals map[string]interface{}, conf *actio
 	}()
 	return release, nil
 }
+
+func InstallChartAsync(ns, name, url string, vals map[string]interface{}, conf *action.Configuration, client dynamic.Interface, coreClient corev1client.CoreV1Interface, fileCleanUp bool, indexEntry string) (*Secret, error) {
```

Comment:
I see a lot of duplicate code, can you consolidate in single function all the duplicate?

Reply:
Agree, method starts completely like `InstallChart`, and an async param makes it also easier to understand the differences.

Reply:
The reason the code is duplicated because we would be removing the sync install/upgrade api after ui transition to using async api's.